core/disenchant.lua

```diff
@@ -81,7 +81,7 @@ function M.value(item_id, slot, quality, level)
 end
 
 function M.distribution(item_id, slot, quality, level)
-    if IGNORE[item_id] or not ARMOR[slot] and not WEAPON[slot] or level == 0 then
+    if not item_id or IGNORE[item_id] or not ARMOR[slot] and not WEAPON[slot] or level == 0 then
```

Comment:
Why would it ever be called with an undefined item id? I think it should be fixed there rather than here.

Reply:
This check is additional protection against improper functioning. If something in the addon will work incorrectly and the first argument will be `nil`, then this check will prevent the code execution below for the case when do not need to do. It is precisely because of the lack of such a check, I immediately did not find this problem that I use an incorrect table in which there is no `item_id` field

Reply:
I see, but I would prefer an assertion then like assert(item_id, "item_id parameter is mandatory")

Reply:
made)